### ISSUE TITLE
Missing ElementHandle.select; impossible (?) to make selection in a frame

### ISSUE THREAD

**Issue by gajus** [puppeteer/puppeteer#1139]
There is no `frame.select` equivalent of the [`page.select`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageselectselector-values). As far as I can tell, there is no way to select an option within an iframe.

**Comment by gajus** [comment]
As a quick workaround, here is a helper function replicating the [`page.select` behaviour](https://github.com/GoogleChrome/puppeteer/blob/126ab7b90e475daabc9d595f1d3c98a1473eb0de/lib/Page.js#L773-L793):

```js
const select = async (frame, selector, ...values) => {
  await frame.$eval(selector, (element, values) => {
    if (element.nodeName.toLowerCase() !== 'select')
      throw new Error('Element is not a <select> element.');

    const options = Array.from(element.options);

    if (element.multiple) {
      for (const option of options)
        option.selected = values.includes(option.value);
    } else {
      element.value = values.shift();
    }
    element.dispatchEvent(new Event('input', { 'bubbles': true }));
    element.dispatchEvent(new Event('change', { 'bubbles': true }));
  }, values);
}

```

**Comment by gajus** [comment]
In general, considering that Puppeteer wants to stay lean, I would advise getting rid off all built-it methods such as select, click, etc. and provide them as helper functions in the above format.

This will solve the problem that you have at the moment – page and frame methods diverging, e.g. https://github.com/GoogleChrome/puppeteer/issues/1051.

**Comment by aslushnikov** [comment]
I'd be happy to review a PR that adds `select` method to the frame as well.

**Comment by austinkelleher** [comment]
@aslushnikov I fixed this in #1194

**Comment by infosuvratjain** [comment]
@aslushnikov  I fixed this #1198

**Comment by ghost** [comment]
@aslushnikov  #1198 solves  frame.select method
